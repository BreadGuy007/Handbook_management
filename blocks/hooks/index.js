/**
 * WordPress dependencies
 */
import createHooks from '@wordpress/hooks';

/**
 * Internal dependencies
 */
import anchor from './anchor';

const {
	addAction,
	addFilter,
	removeAction,
	removeFilter,
	removeAllActions,
	removeAllFilters,
	doAction,
	applyFilters,
	doingAction,
	doingFilter,
	didAction,
	didFilter,
	hasAction,
	hasFilter,
} = createHooks();

export {
	addAction,
	addFilter,
	removeAction,
	removeFilter,
	removeAllActions,
	removeAllFilters,
	doAction,
	applyFilters,
	doingAction,
	doingFilter,
	didAction,
	didFilter,
	hasAction,
	hasFilter,
};

addFilter( 'registerBlockType', 'core\supports-anchor', anchor );
