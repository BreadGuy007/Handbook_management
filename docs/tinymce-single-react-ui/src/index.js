import React, { createElement, Component } from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import Turducken from './Turducken';
import content from './reducers/content'

const store = createStore(content)

const render = () => ReactDOM.render(
	<Provider store={store}>
		<Turducken myStore={store}/>
	</Provider>
		, document.getElementById('tiny-react')
	);

render()
store.subscribe(render)

// TODO: wrap Turducken in a Provider and add the react-redux stuff
