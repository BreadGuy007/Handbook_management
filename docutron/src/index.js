/**
 * External Dependencies
 */
import React from 'react';
import { render } from 'react-snapshot';

/**
 * User Dependencies
 */
import 'config';

/**
 * Internal Dependencies
 */
import App from './components/app';
import './styles/main.css';

render( <App />, document.getElementById( 'root' ) );
