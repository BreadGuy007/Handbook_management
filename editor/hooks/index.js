/**
 * Internal dependencies
 */
import './copy-content';
