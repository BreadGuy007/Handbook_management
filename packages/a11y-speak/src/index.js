import addContainer from './addContainer';
import clear from './clear';
import domReady from './domReady';

let containerPolite, containerAssertive, previousMessage = "";

/**
 * Create the live regions when the DOM is fully loaded.
 */
domReady( function() {
	containerPolite = document.getElementById( "a11y-speak-polite" );
	containerAssertive = document.getElementById( "a11y-speak-assertive" );

	if ( containerPolite === null ) {
		containerPolite = addContainer( "polite" );
	}
	if ( containerAssertive === null ) {
		containerAssertive = addContainer( "assertive" );
	}
} );

/**
 * Update the ARIA live notification area text node.
 *
 * @param {String} message  The message to be announced by Assistive Technologies.
 * @param {String} ariaLive Optional. The politeness level for aria-live. Possible values:
 *                          polite or assertive. Default polite.
 */
const A11ySpeak = function( message, ariaLive ) {
	// Clear previous messages to allow repeated strings being read out.
	clear();

	/*
	 * Strip HTML tags (if any) from the message string. Ideally, messages should
	 * be simple strings, carefully crafted for specific use with A11ySpeak.
	 * When re-using already existing strings this will ensure simple HTML to be
	 * stripped out and replaced with a space. Browsers will collapse multiple
	 * spaces natively.
	 */
	message = message.replace( /<[^<>]+>/g, " " );

	if ( previousMessage === message ) {
		message = message + "\u00A0";
	}

	previousMessage = message;

	if ( containerAssertive && "assertive" === ariaLive ) {
		containerAssertive.textContent = message;
	} else if ( containerPolite ) {
		containerPolite.textContent = message;
	}
};

export default A11ySpeak;
