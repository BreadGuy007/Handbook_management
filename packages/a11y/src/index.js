/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';

/**
 * Internal dependencies
 */
import addContainer from './addContainer';
import clear from './clear';
import filterMessage from './filterMessage';

/**
 * Create the live regions.
 */
export const setup = function() {
	const containerPolite = document.getElementById( 'a11y-speak-polite' );
	const containerAssertive = document.getElementById(
		'a11y-speak-assertive'
	);

	if ( containerPolite === null ) {
		addContainer( 'polite' );
	}
	if ( containerAssertive === null ) {
		addContainer( 'assertive' );
	}
};

/**
 * Run setup on domReady.
 */
domReady( setup );

/**
 * Allows you to easily announce dynamic interface updates to screen readers using ARIA live regions.
 * This module is inspired by the `speak` function in wp-a11y.js
 *
 * @param {string} message  The message to be announced by Assistive Technologies.
 * @param {string} ariaLive Optional. The politeness level for aria-live. Possible values:
 *                          polite or assertive. Default polite.
 *
 * @example
 * ```js
 * import { speak } from '@wordpress/a11y';
 *
 * // For polite messages that shouldn't interrupt what screen readers are currently announcing.
 * speak( 'The message you want to send to the ARIA live region' );
 *
 * // For assertive messages that should interrupt what screen readers are currently announcing.
 * speak( 'The message you want to send to the ARIA live region', 'assertive' );
 * ```
 */
export const speak = function( message, ariaLive ) {
	// Clear previous messages to allow repeated strings being read out.
	clear();

	message = filterMessage( message );

	const containerPolite = document.getElementById( 'a11y-speak-polite' );
	const containerAssertive = document.getElementById(
		'a11y-speak-assertive'
	);

	if ( containerAssertive && 'assertive' === ariaLive ) {
		containerAssertive.textContent = message;
	} else if ( containerPolite ) {
		containerPolite.textContent = message;
	}
};
