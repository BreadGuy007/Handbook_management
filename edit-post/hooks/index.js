/**
 * Internal dependencies
 */
import './blocks';
import './more-menu';
import './validate-use-once';
