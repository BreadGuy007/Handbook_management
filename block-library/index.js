/**
 * WordPress dependencies
 */
import {
	registerBlockType,
	setDefaultBlockName,
	setUnknownTypeHandlerName,
} from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import './style.scss';
import './editor.scss';
import './theme.scss';
import * as paragraph from '../packages/block-library/src/paragraph';
import * as image from '../packages/block-library/src/image';
import * as heading from '../packages/block-library/src/heading';
import * as quote from '../packages/block-library/src/quote';
import * as gallery from '../packages/block-library/src/gallery';
import * as archives from '../packages/block-library/src/archives';
import * as audio from '../packages/block-library/src/audio';
import * as button from '../packages/block-library/src/button';
import * as categories from '../packages/block-library/src/categories';
import * as code from '../packages/block-library/src/code';
import * as columns from '../packages/block-library/src/columns';
import * as column from '../packages/block-library/src/columns/column';
import * as coverImage from '../packages/block-library/src/cover-image';
import * as embed from '../packages/block-library/src/embed';
import * as file from '../packages/block-library/src/file';
import * as latestComments from '../packages/block-library/src/latest-comments';
import * as latestPosts from '../packages/block-library/src/latest-posts';
import * as list from '../packages/block-library/src/list';
import * as more from '../packages/block-library/src/more';
import * as nextpage from '../packages/block-library/src/nextpage';
import * as preformatted from '../packages/block-library/src/preformatted';
import * as pullquote from '../packages/block-library/src/pullquote';
import * as reusableBlock from '../packages/block-library/src/block';
import * as separator from '../packages/block-library/src/separator';
import * as shortcode from '../packages/block-library/src/shortcode';
import * as spacer from '../packages/block-library/src/spacer';
import * as subhead from '../packages/block-library/src/subhead';
import * as table from '../packages/block-library/src/table';
import * as textColumns from '../packages/block-library/src/text-columns';
import * as verse from '../packages/block-library/src/verse';
import * as video from '../packages/block-library/src/video';

// The freeform block can't be moved to the "npm" packages folder because it requires the wp.oldEditor global.
import * as freeform from './freeform';

// The HTML block can't be moved to the "npm" packages folder because it requires the CodeEditor component.
import * as html from './html';

export const registerCoreBlocks = () => {
	[
		// Common blocks are grouped at the top to prioritize their display
		// in various contexts â€” like the inserter and auto-complete components.
		paragraph,
		image,
		heading,
		gallery,
		list,
		quote,

		// Register all remaining core blocks.
		shortcode,
		archives,
		audio,
		button,
		categories,
		code,
		columns,
		column,
		coverImage,
		embed,
		...embed.common,
		...embed.others,
		file,
		freeform,
		html,
		latestComments,
		latestPosts,
		more,
		nextpage,
		preformatted,
		pullquote,
		separator,
		reusableBlock,
		spacer,
		subhead,
		table,
		textColumns,
		verse,
		video,
	].forEach( ( { name, settings } ) => {
		registerBlockType( name, settings );
	} );

	setDefaultBlockName( paragraph.name );
	setUnknownTypeHandlerName( freeform.name );
};
