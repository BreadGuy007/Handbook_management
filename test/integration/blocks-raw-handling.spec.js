/**
 * External dependencies
 */
import fs from 'fs';
import path from 'path';

/**
 * WordPress dependencies
 */
import {
	getBlockContent,
	pasteHandler,
	rawHandler,
	serialize,
} from '@wordpress/blocks';
import { registerCoreBlocks } from '@wordpress/block-library';

function readFile( filePath ) {
	return fs.existsSync( filePath ) ? fs.readFileSync( filePath, 'utf8' ).trim() : '';
}

describe( 'Blocks raw handling', () => {
	beforeAll( () => {
		// Load all hooks that modify blocks
		require( '../../packages/editor/src/hooks' );
		registerCoreBlocks();
	} );

	it( 'should filter inline content', () => {
		const filtered = pasteHandler( {
			HTML: '<h2><em>test</em></h2>',
			mode: 'INLINE',
		} );

		expect( filtered ).toBe( '<em>test</em>' );
		expect( console ).toHaveLogged();
	} );

	it( 'should parse Markdown', () => {
		const filtered = pasteHandler( {
			HTML: '* one<br>* two<br>* three',
			plainText: '* one\n* two\n* three',
			mode: 'AUTO',
		} ).map( getBlockContent ).join( '' );

		expect( filtered ).toBe( '<ul><li>one</li><li>two</li><li>three</li></ul>' );
		expect( console ).toHaveLogged();
	} );

	it( 'should parse inline Markdown', () => {
		const filtered = pasteHandler( {
			HTML: 'Some **bold** text.',
			plainText: 'Some **bold** text.',
			mode: 'AUTO',
		} );

		expect( filtered ).toBe( 'Some <strong>bold</strong> text.' );
		expect( console ).toHaveLogged();
	} );

	it( 'should parse HTML in plainText', () => {
		const filtered = pasteHandler( {
			HTML: '&lt;p&gt;Some &lt;strong&gt;bold&lt;/strong&gt; text.&lt;/p&gt;',
			plainText: '<p>Some <strong>bold</strong> text.</p>',
			mode: 'AUTO',
		} );

		expect( filtered ).toBe( 'Some <strong>bold</strong> text.' );
		expect( console ).toHaveLogged();
	} );

	it( 'should parse Markdown with HTML', () => {
		const filtered = pasteHandler( {
			HTML: '',
			plainText: '# Some <em>heading</em>\n\nA paragraph.',
			mode: 'AUTO',
		} ).map( getBlockContent ).join( '' );

		expect( filtered ).toBe( '<h1>Some <em>heading</em></h1><p>A paragraph.</p>' );
		expect( console ).toHaveLogged();
	} );

	it( 'should break up forced inline content', () => {
		const filtered = pasteHandler( {
			HTML: '<p>test</p><p>test</p>',
			mode: 'INLINE',
		} );

		expect( filtered ).toBe( 'test<br>test' );
		expect( console ).toHaveLogged();
	} );

	it( 'should normalize decomposed characters', () => {
		const filtered = pasteHandler( {
			HTML: 'schön',
			mode: 'INLINE',
		} );

		expect( filtered ).toBe( 'schön' );
		expect( console ).toHaveLogged();
	} );

	it( 'should treat single list item as inline text', () => {
		const filtered = pasteHandler( {
			HTML: '<ul><li>Some <strong>bold</strong> text.</li></ul>',
			plainText: 'Some <strong>bold</strong> text.\n',
			mode: 'AUTO',
		} );

		expect( filtered ).toBe( 'Some <strong>bold</strong> text.' );
		expect( console ).toHaveLogged();
	} );

	it( 'should treat multiple list items as a block', () => {
		const filtered = pasteHandler( {
			HTML: '<ul><li>One</li><li>Two</li><li>Three</li></ul>',
			plainText: 'One\nTwo\nThree\n',
			mode: 'AUTO',
		} ).map( getBlockContent ).join( '' );

		expect( filtered ).toBe( '<ul><li>One</li><li>Two</li><li>Three</li></ul>' );
		expect( console ).toHaveLogged();
	} );

	describe( 'pasteHandler', () => {
		[
			'plain',
			'classic',
			'apple',
			'google-docs',
			'ms-word',
			'ms-word-styled',
			'ms-word-online',
			'evernote',
			'iframe-embed',
			'one-image',
			'two-images',
			'markdown',
			'wordpress',
			'gutenberg',
			'caption-shortcode',
		].forEach( ( type ) => {
			it( type, () => {
				const HTML = readFile( path.join( __dirname, `fixtures/${ type }-in.html` ) );
				const plainText = readFile( path.join( __dirname, `fixtures/${ type }-in.txt` ) );
				const output = readFile( path.join( __dirname, `fixtures/${ type }-out.html` ) );
				const converted = pasteHandler( { HTML, plainText, canUserUseUnfilteredHTML: true } );
				const serialized = typeof converted === 'string' ? converted : serialize( converted );

				expect( serialized ).toBe( output );

				if ( type !== 'gutenberg' ) {
					expect( console ).toHaveLogged();
				}
			} );
		} );
	} );
} );

describe( 'rawHandler', () => {
	it( 'should convert HTML post to blocks with minimal content changes', () => {
		const HTML = readFile( path.join( __dirname, 'fixtures/wordpress-convert.html' ) );
		expect( serialize( rawHandler( { HTML } ) ) ).toMatchSnapshot();
	} );

	it( 'should convert a caption shortcode', () => {
		const HTML = readFile( path.join( __dirname, 'fixtures/shortcode-caption.html' ) );
		expect( serialize( rawHandler( { HTML } ) ) ).toMatchSnapshot();
	} );

	it( 'should convert a caption shortcode with link', () => {
		const HTML = readFile( path.join( __dirname, 'fixtures/shortcode-caption-with-link.html' ) );
		expect( serialize( rawHandler( { HTML } ) ) ).toMatchSnapshot();
	} );

	it( 'should convert a caption shortcode with caption', () => {
		const HTML = readFile( path.join( __dirname, 'fixtures/shortcode-caption-with-caption-link.html' ) );
		expect( serialize( rawHandler( { HTML } ) ) ).toMatchSnapshot();
	} );
} );
