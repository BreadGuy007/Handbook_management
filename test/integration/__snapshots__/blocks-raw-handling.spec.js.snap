// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rawHandler should convert HTML post to blocks with minimal content changes 1`] = `
"<!-- wp:heading -->
<h2>Howdy</h2>
<!-- /wp:heading -->

<!-- wp:image -->
<figure class=\\"wp-block-image\\"><img src=\\"https://w.org\\" alt=\\"\\"/></figure>
<!-- /wp:image -->

<!-- wp:paragraph -->
<p>This is a paragraph.</p>
<!-- /wp:paragraph -->

<!-- wp:paragraph -->
<p>Preserve <span style=\\"color:red\\">me</span>!</p>
<!-- /wp:paragraph -->

<!-- wp:heading {\\"level\\":3} -->
<h3>More tag</h3>
<!-- /wp:heading -->

<!-- wp:more -->
<!--more-->
<!-- /wp:more -->

<!-- wp:heading {\\"level\\":3} -->
<h3>Shortcode</h3>
<!-- /wp:heading -->

<!-- wp:gallery {\\"ids\\":[1],\\"columns\\":3,\\"linkTo\\":\\"attachment\\"} -->
<ul class=\\"wp-block-gallery columns-3 is-cropped\\"><li class=\\"blocks-gallery-item\\"><figure><img data-id=\\"1\\" class=\\"wp-image-1\\"/></figure></li></ul>
<!-- /wp:gallery -->

<!-- wp:html -->
<dl>
	<dt>Term</dt>
	<dd>
		Description.
	</dd>
</dl>
<!-- /wp:html -->

<!-- wp:list {\\"ordered\\":true} -->
<ol><li>Item</li></ol>
<!-- /wp:list -->

<!-- wp:quote -->
<blockquote class=\\"wp-block-quote\\"><p>Text.</p></blockquote>
<!-- /wp:quote -->

<!-- wp:html -->
<blockquote><h1>Heading</h1><p>Text.</p></blockquote>
<!-- /wp:html -->"
`;

exports[`rawHandler should convert a caption shortcode 1`] = `
"<!-- wp:image {\\"id\\":122,\\"align\\":\\"none\\",\\"className\\":\\"size-medium wp-image-122\\"} -->
<figure class=\\"wp-block-image alignnone size-medium wp-image-122\\"><img src=\\"image.png\\" alt=\\"\\" class=\\"wp-image-122\\"/><figcaption>test</figcaption></figure>
<!-- /wp:image -->"
`;

exports[`rawHandler should convert a caption shortcode with caption 1`] = `
"<!-- wp:image {\\"id\\":122,\\"align\\":\\"none\\",\\"className\\":\\"size-medium wp-image-122\\"} -->
<figure class=\\"wp-block-image alignnone size-medium wp-image-122\\"><img src=\\"image.png\\" alt=\\"\\" class=\\"wp-image-122\\"/><figcaption><a href=\\"https://w.org\\">test</a></figcaption></figure>
<!-- /wp:image -->"
`;

exports[`rawHandler should convert a caption shortcode with link 1`] = `
"<!-- wp:image {\\"id\\":754,\\"align\\":\\"none\\"} -->
<figure class=\\"wp-block-image alignnone\\"><a href=\\"http://build.wordpress-develop.test/wp-content/uploads/2011/07/100_5478.jpg\\"><img src=\\"http://build.wordpress-develop.test/wp-content/uploads/2011/07/100_5478.jpg?w=604\\" alt=\\"Bell on Wharf\\" class=\\"wp-image-754\\"/></a><figcaption>Bell on wharf in San Francisco</figcaption></figure>
<!-- /wp:image -->"
`;
